compiler/grainformat/walktree.re

```diff
@@ -119,10 +119,22 @@ let walktree =
 
   all_locations := comment_locations;
 
-  let iter_location = (self, location) =>
-    if (!List.mem(Code(location), all_locations^)) {
-      all_locations := List.append(all_locations^, [Code(location)]);
+  // use a hash table to de-deupe adding locations to the list
+  // because the iterator returns the same location multiple times
+
+  // best guess of size, 20 nodes per top level stmt
```

Comment:
Probably more efficient to just guess a total size. Something like a static 4096 probably works fine here.